import "goby/common/protobuf/option_extensions.proto";
import "dccl/protobuf/option_extensions.proto";

package goby.acomms.protobuf;

message ModemDriverStatus
{
    option (dccl.msg).id = 0x0201;
    option (dccl.msg).max_bytes = 32;

    required int32 src = 1 [
        (dccl.field).min = 1,
        (dccl.field).max = 32,
        (dccl.field).in_head = true
    ];
    optional int32 dest = 2 [
        (dccl.field).min = 1,
        (dccl.field).max = 32,
        (dccl.field).in_head = true
    ];

    required double time = 3
        [(dccl.field).codec = "_time", (dccl.field).in_head = true];

    enum Status
    {
        NOMINAL = 0;
        CONNECTION_TO_MODEM_FAILED = 1;
        MODEM_NOT_RESPONDING = 2;
        STARTUP_FAILED = 3;
        INVALID_CONFIGURATION = 4;
        MANUAL_RESET = 5;
    }

    required Status status = 4;

    optional uint32 n_resets = 5
        [default = 0, (dccl.field).min = 0, (dccl.field).max = 10000];
}
